### PR TITLE
Introduce filter-bar styles

### PR DESCRIPTION
**Pull request by daiyiwhale**
Creating specific styles for filter-bar situations.

### CODE REVIEW

File: src/styles/components/filter-bar.less [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+    }
+
+    & > .filter-bar-item {
+
+      &:last-child {
+
+        // This overrides all of the specificity applied by the classes
+        // below.
+        padding-right: 0 !important;
```
Comment (orlandohohmeier) [inline src/styles/components/filter-bar.less]: If I don't miss anything special, it should just work without the `!important`. Besides this, I'd like you to simplify the selector and just use `.filter-bar-item:last-child` instead as there is no need for the extra `& >`.

**Example**
```less
 .filter-bar-right {

    margin-left: auto;

    &:last-child {

      padding-right: 0;

    }

    .filter-bar-item:last-child {

      // This overrides all of the specificity applied by the classes
      // below.
      padding-right: 0;

    }

  }
```

**Comment by kennyt** [comment]
does it apply to the side panel task view too? (the spacing at the bottom of that is pretty tall)

px

**Comment by mlunoe** [comment]
I'm not quite sure, but maybe we should use the same thing for the TaskDebugView too:


This could of course make sense to do in a separate PR

File: src/styles/components/filter-bar.less (hunk 1 above)
Comment (daiyiwhale) [inline src/styles/components/filter-bar.less]: Good idea, thanks!

**Comment by daiyiwhale** [comment]
I was going to do the side panel ones in a separate PR since the side panels are being nuked anyway

**Comment by kennyt** [comment]
side panels might be getting nuked, but the contents are only going to be moved right? so it'll still matter

but i'm cool with it being in another PR. just do it soon so we don't forget about it :)

**Comment by daiyiwhale** [comment]
I'm also nuking the node side panels this week and didn't want to give myself rebase problems haha :P I won't forget

File: src/styles/components/filter-bar.less [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+      padding-right: 0;
+
+    }
+
+    .filter-bar-item:last-child {
+
+      // This overrides all of the specificity applied by the classes
+      // below.
+      padding-right: 0 !important;
```
Comment (orlandohohmeier) [inline src/styles/components/filter-bar.less]: I'm sorry to bug you once again, and I'm probably missing something obvious, but I would really love to get rid of the `!important` statement.

filter item important statement

At least in Chrome it seems to work just fine without the `!important` statement. Could you please give it another try?

Comment (daiyiwhale) [inline src/styles/components/filter-bar.less]: Ah, I totally meant to make this change but overlooked the `!important` word; making this change now and testing on all browsers!

Comment (daiyiwhale) [inline src/styles/components/filter-bar.less]: Alright it looks fine on Chrome, FF, Safari, and Edge!

Comment (daiyiwhale) [inline src/styles/components/filter-bar.less]: Interesting finding: filter search input doesn't expand to full width on Safari and FF at mobile view breakpoint:

Comment (daiyiwhale) [inline src/styles/components/filter-bar.less]: Sorry to have missed this the first time @orlandohohmeier, thanks for your patience and persistence! (:

File: src/js/pages/system/UnitsHealthTab.js [hunk 3]
```diff
@@ -165,33 +165,31 @@ class UnitsHealthTab extends mixin(StoreMixin) {
...
-                    handleFilterChange={this.handleSearchStringChange}
-                    inverseStyle={true} />
-                </div>
+          <div className="filter-bar">
+            <div className="filter-bar-left">
+              <div className="filter-bar-item">
+                <FilterButtons
+                  renderButtonContent={this.getButtonContent}
+                  filters={['all', 'healthy', 'unhealthy']}
+                  filterByKey={'title'}
```
Comment (kennyt) [inline src/js/pages/system/UnitsHealthTab.js]: if it's just a string then we can do `="title"` :)

Comment (daiyiwhale) [inline src/js/pages/system/UnitsHealthTab.js]: good call though, I'll nix it in the PR for using `FilterBar` component

**Pull request merged**